### DISCUSSION TITLE
RunContext.get().createSpecRunner(spec, notifier).run(); alternative in 2.0

### DISCUSSION THREAD

**Discussion by Morvell** [spockframework/spock#1485] [Q&A]
Hello. In version 1.3, you can start the context with the command 
`
RunNotifier notifier = new RunNotifier();
SpecInfo spec = new SpecInfoBuilder(clazz).build();
RunContext.get().createSpecRunner(spec, notifier).run();
`

what is the equivalent in version 2?

**Comment by leonard84** [comment]
What exactly do you want to achieve? 

If you want to execute Spock programmatically have a look at the [junit platform launcher api](https://junit.org/junit5/docs/current/user-guide/#launcher-api-execution)

**Reply by Morvell** [reply]
I need to test the implementation IGlobalExtension

**Reply by leonard84** [reply]
Have a look at the Spock code base, for example https://github.com/spockframework/spock/blob/53ae1b2d3d774b62b7c7ff5ceb21c321ea5ab1e3/spock-specs/src/test/groovy/org/spockframework/smoke/extension/IncludeExcludeFeatures.groovy which is a test for the https://github.com/spockframework/spock/blob/53ae1b2d3d774b62b7c7ff5ceb21c321ea5ab1e3/spock-core/src/main/java/org/spockframework/runtime/extension/builtin/IncludeExcludeExtension.java 
You basically use the `EmbeddedSpecRunner` to run a spec in a spec. Keep in mind that you need to add the `junit-platform-testkit` dependency for it to work.
